src/python_tools/sensor_proto_log.py

```diff
@@ -0,0 +1,33 @@
+from .proto_log import ProtoLog
+from software.proto.sensor_msg_pb2 import SensorProto
+
+
+class SensorProtoLog:
+    def __init__(self, directory):
+        proto_log = ProtoLog(directory, SensorProto)
+        self._messages = dict()
+        for sensor_msg in proto_log:
+            # ListFields() gives a list of tuple of (FieldDescriptor, Message)
+            field_names = [field[0].name for field in sensor_msg.ListFields()]
+
+            # iterate over field names for each message and add them to the dict entry
+            # for that particular field
+            for field_name in field_names:
+                if field_name != "backend_received_time":
+                    if field_name not in self._messages:
+                        self._messages[field_name] = []
+                    self._messages[field_name].append(getattr(sensor_msg, field_name))
```

Comment:
As a more general thought, isn't there a proto library for interfacing with proto files? What was the rationale for converting them into a dict like this?

Reply:
Because `SensorProto` is a message with a bunch of optional fields of different message types (which are the messages we are *actually* interested in), it makes sense to unpack and 'demux' them to make it easier to work with the data.